FormalConjectures/ErdosProblems/317.lean

```diff
@@ -0,0 +1,68 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+
+/-!
+# Erdős Problem 317
+
+*Reference:* [erdosproblems.com/317](https://www.erdosproblems.com/317)
+-/
+
+namespace Erdos317
+
+/--
+The least common multiple of the numbers 1,...,n.
+-/
+abbrev lcm_one_to_n : ℕ → ℕ
+    | 0 => 1
+    | n + 1 => (lcm_one_to_n n).lcm (n + 1)
+
+/--
+Inequality is obvious for `erdos_317.variants.claim2`, the problem is strict inequality.
+This fails for small $n$, for example\[\frac{1}{2}-\frac{1}{3}-\frac{1}{4}=-\frac{1}{12}.\]
+-/
+@[category API, AMS 11]
+lemma claim2_inequality : ∃ N : ℕ, ∀ n ≥ N,
+    ∀ δ : ℕ → ℝ, δ '' (Finset.Icc 1 n) ⊆ {-1,0,1} →
+    abs (∑ k ∈ Finset.Icc 1 n, ((δ k : ℝ) / k)) ≠ 0 →
+        abs (∑ k ∈ Finset.Icc 1 n, ((δ k : ℝ) / k)) ≥ 1 / (lcm_one_to_n n : ℝ) := by
```

Comment:
```suggestion
    letI lhs := |∑ k ∈ Finset.Icc 1 n, ((δ k : ℝ) / k)|
    lhs  ≠ 0 → lhs ≥ 1 / (lcm_one_to_n n : ℝ) := by
```

Consider this inline definition, so that you don't have to repeat defining the sum. Also notation `|...|` might be nicer than `abs`

Here and everywhere below